src/Sql/Sql/TransparentDataEncryption/Model/AzureRmSqlManagedInstanceKeyVaultKeyModel.cs

```diff
@@ -0,0 +1,98 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Microsoft.Azure.Management.Sql.Models;
+
+namespace Microsoft.Azure.Commands.Sql.TransparentDataEncryption.Model
+{
+    /// <summary>
+    /// Represents the core properties of an Azure Sql Database Transparent Data Encryption Certificate
+    /// </summary>
+    public class AzureRmSqlManagedInstanceKeyVaultKeyModel
```

Comment:
consider making a base class that is shared with corresponding DB class.

Reply:
This does not seem to be standard pattern and might require refactoring other classes as well. I think we should decouple this from the current workitem as it is blocking MI GA